CharlieBackend.Data/Repositories/Impl/DashboardRepository.cs

```diff
@@ -0,0 +1,396 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using CharlieBackend.Core.Entities;
+using Microsoft.EntityFrameworkCore;
+using CharlieBackend.Core.DTO.Dashboard;
+using CharlieBackend.Core.DTO.StudentGroups;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+
+namespace CharlieBackend.Data.Repositories.Impl
+{
+    public class DashboardRepository : Repository<StudentGroup>, IDashboardRepository
+    {
+        public DashboardRepository(ApplicationContext applicationContext)
+                : base(applicationContext)
+        {
+        }
+
+        public async Task<List<long>> GetGroupsIdsByCourceIdAsync(long courceId, DateTime startDate, DateTime finishDate)
```

Comment:
It is not clear from method signature, that it can work ok with empty dates. Either make them nullable and with default null value, or leave them as is and remove if (startDate == default(DateTime) && finishDate == default(DateTime))

Reply:
As in our existing database, START and FINISH course period fields can be null, I left in code this part ` if (startDate == default(DateTime) && finishDate == default(DateTime)`

Reply:
Maybe for better obviousness better write method overload?

Reply:
Ok. Implemented ".WhereIf" extension method to simplify code